roboteam_ai/src/control/positionControllers/ForcePosControl.cpp

```diff
@@ -25,12 +26,12 @@ Vector2 ForcePosControl::calculateForces(const RobotPtr &robot, const Vector2 &t
     force = (force.length() > 3.0) ? force.stretchToLength(3.0) : force;
 
     // avoid our own robots
-    for (auto bot : world.us) {
+    for (auto &bot : world.us) {
```

Comment:
can this function also not be const? I think all the operations on it are const atleast.

Reply:
auto does that for you

Reply:
Nope, const auto & has different behavior than auto & does and makes sure you avoid editing your own robots or robotpointers on accident.

Reply:
Make it const